File: CompanyApi/DTOs/CreateCompanyDto.cs
using System.ComponentModel.DataAnnotations;

namespace CompanyApi.DTOs
{
    public class CreateCompanyDto
    {
        [Required]
        public string Name { get; set; }
        
        [Required]
        public string Exchange { get; set; }
        
        [Required]
        public string Ticker { get; set; }

        [Required]
        [MinLength(12)]
        [MaxLength(12)]
        [RegularExpression(@"^[^\d]{2}.*$", ErrorMessage = "ISIN must start with two non-numeric characters.")]
        public string Isin { get; set; }
        
        public string? Website { get; set; }
    }
}

File: company-frontend/src/app/domains/companies/models/create-company-dto.ts
import { CreateCompanyDto as CreateCompanyApiDto } from '@api/models/create-company-dto';
import { Company } from './company';

export interface CreateCompanyDto {
  exchange: string;
  isin: string;
  name: string;
  ticker: string;
  website?: string;
}

export function mapCreateCompanyDtoToApi(
  createCompanyDto: CreateCompanyDto
): CreateCompanyApiDto {
  return {
    exchange: createCompanyDto.exchange,
    isin: createCompanyDto.isin,
    name: createCompanyDto.name,
    ticker: createCompanyDto.ticker,
    website: createCompanyDto.website,
  };
}

export function mapCompanyToCreateCompanyDto(
  createCompanyDto: Company
): CreateCompanyDto {
  return {
    exchange: createCompanyDto.exchange,
    isin: createCompanyDto.isin,
    name: createCompanyDto.name,
    ticker: createCompanyDto.ticker,
    website: createCompanyDto.website ?? undefined,
  };
}
